tests/nightly/test_large_array.py

```diff
@@ -1350,6 +1351,16 @@ def run_trsm(inp):
     check_batch_trsm()
 
 
+def test_linalg_errors():
+    def check_syevd_error():
+        A = get_identity_mat(LARGE_SQ_X)
+        for i in range(LARGE_SQ_X):
+            A[i,i] = 1
+        assertRaises(MXNetError, mx.nd.linalg.syevd, A)
```

Comment:
I think the error is raised when the operator is executed rather than when it's pushed. So this assertion will not catch an error. 

```
def test_linalg_large_dim():
    def check_gemm():
        A = mx.nd.ones(shape=(1, 2**31))
        B = mx.nd.ones(shape=(1, 2**31))
        C = mx.nd.ones(shape=(1,1))
        D = mx.nd.linalg.gemm(A, B, C, transpose_b=True, alpha=2.0 , beta=10.0)
        nd.waitall()
    assertRaises(MXNetError, check_gemm)
```

This worked for me. We need to wrap an additional layer and call nd.waitall() in that layer